### PR TITLE
adding check for bandit vulnerabilities 1/n

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

validated with `flake8-bandit`
leaving a few skipped with todo

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

**Review by carmocca — COMMENTED** [review]
Can you explain what this does? Why is this change useful?

Seems like there are many false positives

**Comment by awaelchli** [comment]
Can you point out the vulnerability? It is not clear to me where the fix is

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -67,6 +67,10 @@ repos:
     hooks:
       - id: yesqa
         name: Unused noqa
+        additional_dependencies:
+          #- pep8-naming
+          #- flake8-pytest-style
+          - flake8-bandit
```
Comment (Borda) [inline .pre-commit-config.yaml:73]: This was removing added ignores 105 106 107

**Comment by Borda** [comment]
> Can you point out the vulnerability? It is not clear to me where the fix is

Not many are fixed, and you are right that this is polluted with fixes of #17386
so, in fact, this rather setting the sage so no more shall be added, and let's fix these later one by one... :chipmunk:

File: examples/fabric/build_your_own_trainer/run.py [hunk 2]
```diff
@@ -58,8 +58,8 @@ def train(model):
     from torchvision.datasets import MNIST
     from torchvision.transforms import ToTensor
 
-    train_set = MNIST(root="/tmp/data/MNIST", train=True, transform=ToTensor(), download=True)
-    val_set = MNIST(root="/tmp/data/MNIST", train=False, transform=ToTensor(), download=False)
+    train_set = MNIST(root="/tmp/data/MNIST", train=True, transform=ToTensor(), download=True)  # noqa: S108
+    val_set = MNIST(root="/tmp/data/MNIST", train=False, transform=ToTensor(), download=False)  # noqa: S108
```
Comment (awaelchli) [inline examples/fabric/build_your_own_trainer/run.py:62]: We need to add an exception rule for the examples, these shouldn't have the noqa notes

File: pyproject.toml [hunk 3]
```diff
@@ -67,6 +68,29 @@ exclude = [
 ]
 ignore-init-module-imports = true
 
+[tool.ruff.per-file-ignores]
+".actions/*" = ["S101", "S310"]
+"setup.py" = ["S101"]
+"examples/**" = [
+    "S101",
+    "S113",  # todo: Probable use of requests call without timeout
+]
```
Comment (awaelchli) [inline pyproject.toml:77]: I recommend ignoring all codes

**Review by awaelchli — APPROVED** [review]
I recommend renaming the PR since this all looks like formatting and there aren't any actual vulnerabilities in the code

File: pyproject.toml (hunk 3 above)
Comment (Borda) [reply pyproject.toml:80]: Do you mean to exclude examples from code quality checks?

Comment (awaelchli) [reply pyproject.toml:81]: I mean not forcing a particular style or special annotations like `#noqa` into example code.

**Review by lantiga — APPROVED** [review]

**Pull request merged**